### ISSUE TITLE
MFA report numbers should be more like a funnel

### ISSUE THREAD

**Issue by brendanheywood** [catalyst/moodle-tool_mfa#176]
/admin/tool/mfa/factor_report.php

ie the first line might look something like:

. | Non-administrator | IP Range Factor | Authenticator app | Grace period
-- | -- | -- | -- | --
manual | 700 | 700 | 20 | 700

But what we want it to show is really the funnel of which uses hit and passed each step, so probably more like:

. | Users | Logins | Non-administrator | IP Range Factor | Authenticator app | Grace period
-- | -- | -- | -- | -- | -- | --
manual | 1000 | 722 | 700 | 20 | 13 | 7

ie in plain english:

1) 720 manual users have attempted to login
2) 700 of them passed the no-admin factor and so stopped there
3) which means 22 were admins, of those 20 logged in via a secure network at some point
4) of those 13 have setup totp
5) of the remained 7 have been in grace mode

So a few tasks might drop out of this:

* [x] have a button to reset the stats
* [x] show the time at which stats have started being recorded from
* [x] add the starting 2 columns which is the raw number users and logins before we do any factor filtering funnel
* [ ] some stats need to be have the ability to fade away, ie the grace mode should be able to remove people whose grace mode has expired. we want that number to be zero after the roll out
* [x] the report itself and the stats should be recording stats on just the passed / triggered factors not all of them
* [ ] have a drill down report so we can see what users are behind each number

**Comment by Peterburnett** [comment]
Spun factor clearing out into #178 
Spun factor drill down reports into #179
